src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -447,40 +447,50 @@ def _create_zone():
                 )
             )
 
-        def _update_zone(current, new):
+        def _update_zone(new):
             new.spec.gateway.instances = spec.placement.count or 1
             return new
 
         return self._create_or_patch(
             cos.CephObjectStore, 'cephobjectstores', name,
             _update_zone, _create_zone)
 
-    def add_nfsgw(self, spec):
+    def apply_nfsgw(self, spec):
+        # type: (ServiceSpec) -> None
         # TODO use spec.placement
         # TODO warn if spec.extended has entries we don't kow how
         #      to action.
+        cnt = spec.placement.count or 1
```

Comment:
should we ensure the `PlacementSpec` always has either a count or a list of hosts?

Reply:
I would prefer not changing it in this PR. As validation of hosts needs to be done in other places too.

Reply:
mind updating the TODO comment to remind us to add pod affinities based on the list of hosts in the PlacementSpec?

Reply:
Done!